### PR TITLE
Adds transparency public functions

### PR DESCRIPTION
**Pull request by wpcarro**
Adds public functions for manipulating HSLA colors. Tests are included as well.

These functions are analogous to the existing functions that modulate saturation 
and lightness.

### CODE REVIEW

File: src/garden/color.cljc [hunk 1]
```diff
@@ -205,6 +211,9 @@
 (def percent-clip
   (partial util/clip 0 100))
 
+(def zero-to-one-clip
```
Comment (noprompt) [inline src/garden/color.cljc]: I know the `partial` function one below doesn't have this but if you could add
```clj
^{:arglists '([n])
  :private true}
zero-to-one-clip
```
to this symbol I'd be very appreciative.

File: src/garden/color.cljc [hunk 2]
```diff
@@ -237,6 +246,15 @@
    (number? x) (hsl [x (percent-clip x) (percent-clip x)])
    :else (throw (ex-info (str "Can't convert " x " to a color.") {}))))
 
+(defn as-hsla
+  "Converts a color to HSLA. Assumes an alpha value of 1.00 unless one is
+  currently set on color."
+  [color]
+  (let [current-alpha (or (:alpha color) 1.00)]
```
Comment (noprompt) [inline src/garden/color.cljc]: `(get color :alpha 1.00)`

File: src/garden/color.cljc [hunk 3]
```diff
@@ -237,6 +246,15 @@
...
+    (cond
```
Comment (noprompt) [inline src/garden/color.cljc]: Just use `if` here.

**Comment by noprompt** [comment]
Thanks for your effort on this. If you could address the remarks I've made that'd be appreciated.

**Comment by wpcarro** [comment]
@noprompt addressed your comments. All fair points imo, so thank you for your time!

**Comment by noprompt** [comment]
@wpcarro Sorry for letting this ferment. I appreciate your contribution!

**Pull request merged**